### DISCUSSION TITLE
Flutter + flutter_hooks v Flutter explanation

### DISCUSSION THREAD

**Discussion by RukshanJS** [rrousselGit/riverpod#2437] [Q&A] [answered]
In the https://riverpod.dev/docs/getting_started#installing-the-package page, there's Flutter + flutter_hooks and Flutter versions of the code for a hello world example. 

But I don't see any large difference between the two codes except for the superclass from which the `MyApp` class extends. It's been a while since I used riverpod and truly loved the useProvider pattern which is sadly not anymore used. 

Could someone please explain why the code snippets in the above linked page looks nearly identical?

**Comment by AhmedLSayed9** [comment, marked as answer]
- If you're a `flutter_hooks` user, you need to extend `HookWidget` to use a hook inside some widget.
- If you're a `flutter_riverpod` user, you need to extend `ConsumerWidget` to get a (ref) to use riverpod inside some widget.
- What If you want to use both in some widget? That's when you'll need to extend `HookConsumerWidget` which comes from `hooks_riverpod` package.

The example might need to be updated to use some hook for more clarity, something like:
```dart
class MyApp extends HookConsumerWidget {
  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final String value = ref.watch(helloWorldProvider);
    final controller = useTextEditingController(text: 'initial text');

    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(title: const Text('Example')),
        body: Center(
          child: Text(value),
        ),
      ),
    );
  }
}
```

Note: when you import `hooks_riverpod` you don't need to import `flutter_riverpod` too, imagine `hooks_riverpod` as a wrapper above `flutter_riverpod` that let you use `HookConsumerWidget` in addition to `ConsumerWidget`. but You'll need to import `flutter_hooks` with it to be able to use hooks.

**Reply by RukshanJS** [reply]
Thanks. I'll try like this and post an update here. Should this be an issue since the documentation is not updated?

@AhmedLSayed9 would you please explain the `controller` and `useTextEditingController` and their usage or point me to somewhere this is explained? Appreciate a lot. thanks again!

**Reply by AhmedLSayed9** [reply]
You can contribute if you want but It's not a big deal. if you're a Flutter Hooks user you should know the idea and there's already a table describe the difference at `What package to install` section.

`useTextEditingController` from https://pub.dev/packages/flutter_hooks

**Reply by RukshanJS** [reply]
Appreciate your reply, and thanks for the link. I see the documentation is in-progress so the minor discrepancies are acceptable IMO. Thanks again.
